src/tim/vx/platform/grpc/remote_service_server.cc

```diff
@@ -0,0 +1,368 @@
+/****************************************************************************
+*
+*    Copyright (c) 2023 Vivante Corporation
+*
+*    Permission is hereby granted, free of charge, to any person obtaining a
+*    copy of this software and associated documentation files (the "Software"),
+*    to deal in the Software without restriction, including without limitation
+*    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+*    and/or sell copies of the Software, and to permit persons to whom the
+*    Software is furnished to do so, subject to the following conditions:
+*
+*    The above copyright notice and this permission notice shall be included in
+*    all copies or substantial portions of the Software.
+*
+*    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+*    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+*    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+*    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+*    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+*    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+*    DEALINGS IN THE SOFTWARE.
+*
+*****************************************************************************/
+#include <grpc/grpc.h>
+#include <grpcpp/security/server_credentials.h>
+#include <grpcpp/server.h>
+#include <grpcpp/server_builder.h>
+#include <grpcpp/server_context.h>
+
+#include "remote_service.grpc.pb.h"
+#include "tim/vx/platform/native.h"
+#include "vsi_nn_pub.h"
+
+std::unordered_map<int32_t, std::shared_ptr<tim::vx::platform::IDevice>>
+    device_table;
+std::unordered_map<int32_t, std::shared_ptr<tim::vx::platform::IExecutor>>
+    executor_table;
+std::vector<std::shared_ptr<tim::vx::platform::IExecutable>> executable_table;
+std::vector<std::shared_ptr<tim::vx::platform::ITensorHandle>> tensor_table;
+
+namespace {
+tim::vx::DataType MapDataType(::rpc::DataType type) {
```

Comment:
duplicate code ?

Reply:
It's not.
we need two type convertor: tim-vx -> gRPC and gPRC->tim-vx